src/mock.js

```diff
@@ -82,6 +82,21 @@ function addMockFunctionsToSchema({ schema, mocks = {}, preserveResolvers = fals
     return customMock;
   }
 
+  function assignResolveType(type) {
+    const fieldType = getNullableType(type);
+    const namedFieldType = getNamedType(fieldType);
+
+    if (namedFieldType instanceof GraphQLUnionType ||
+        namedFieldType instanceof GraphQLInterfaceType
+    ) {
+      const oldResolveType = namedFieldType.resolveType;
+      // the default `resolveType` always returns null. We add a fallback
+      // resolution that works with how unions and interface are mocked
+      namedFieldType.resolveType = (data, context, info) =>
+        oldResolveType(data, context, info) || info.schema.getType(data.typename);
```

Comment:
sorry I have been really busy this past days! Everything you said makes sense. Let me do the required changes. In addition to that, what I was trying to say, is also that in case `preserveResolvers` is true and `field.resolve` is also true, the resolveType will not be mocked, per https://github.com/itajaja/graphql-tools/blob/resolve-type-mock-server/src/mock.js#L172

To avoid that, I can move the mocking of resolve type before that conditional. wdyt?